tests/test_pylogger.py

```diff
@@ -0,0 +1,161 @@
+
+
+class TestDiscoverAppLoggers:
+    """Test cases for _discover_app_loggers function."""
+
+    def setup_method(self):
+        """Reset cache before each test."""
+        # Reset the cache by accessing the module-level variable
+        import template_agent.utils.pylogger as pylogger_module
+
+        pylogger_module._DISCOVERED_LOGGERS_CACHE = None
+
+    def test_discover_loggers_for_real_package(self):
+        """Test discovering loggers for the actual template_agent package."""
+        # Reset cache
+        import template_agent.utils.pylogger as pylogger_module
+
+        pylogger_module._DISCOVERED_LOGGERS_CACHE = None
+
+        # Discover loggers
+        logger_names = _discover_app_loggers("template_agent")
+
+        # Should always include root package
+        assert "template_agent" in logger_names
+
+        # Should find some submodules
+        assert len(logger_names) > 1
+
+        # Should find src and utils subpackages
+        assert any("template_agent.src" in name for name in logger_names)
+        assert any("template_agent.utils" in name for name in logger_names)
+
+    def test_caching_behavior(self):
+        """Test that results are cached after first call."""
+        import template_agent.utils.pylogger as pylogger_module
+
+        pylogger_module._DISCOVERED_LOGGERS_CACHE = None
+
+        # First call
+        first_result = _discover_app_loggers("template_agent")
+
+        # Cache should be populated
+        assert pylogger_module._DISCOVERED_LOGGERS_CACHE is not None
+        assert pylogger_module._DISCOVERED_LOGGERS_CACHE == first_result
+
+        # Second call should return cached result
+        second_result = _discover_app_loggers("template_agent")
+
+        # Should return the same object (not just equal, but same reference)
+        assert second_result is first_result
+
+    def test_fallback_for_nonexistent_package(self):
+        """Test fallback behavior when package cannot be imported."""
+        import template_agent.utils.pylogger as pylogger_module
+
+        pylogger_module._DISCOVERED_LOGGERS_CACHE = None
+
+        # Try to discover loggers for a nonexistent package
+        logger_names = _discover_app_loggers("nonexistent_package")
+
+        # Should include root package name
+        assert "nonexistent_package" in logger_names
+
+        # Should attempt to add common patterns
+        # (they won't actually be added since they don't exist)
+        # But the function should complete without errors
+
+    @patch("importlib.import_module")
+    def test_fallback_patterns_added_if_importable(self, mock_import):
+        """Test that fallback patterns are added if they can be imported."""
+        import template_agent.utils.pylogger as pylogger_module
+
+        pylogger_module._DISCOVERED_LOGGERS_CACHE = None
+
+        # Make the root package import fail, but allow submodules to succeed
+        def side_effect(name):
```

Comment:
`importlib.import_module()` [accepts two parameters](https://docs.python.org/3/library/importlib.html#importlib.import_module), so the mock for it should, too, e.g.,
```suggestion
        # Make the root package import fail, but allow submodules to succeed
        def side_effect(name, **_argv):
```